server/src/main/java/org/apache/kafka/server/ClientMetricsManager.java

```diff
@@ -114,13 +117,15 @@ public ClientMetricsManager(ClientMetricsReceiverPlugin receiverPlugin, int clie
         this.subscriptionMap = new ConcurrentHashMap<>();
         this.subscriptionUpdateVersion = new AtomicInteger(0);
         this.clientInstanceCache = new SynchronizedCache<>(new LRUCache<>(CACHE_MAX_SIZE));
+        this.clientConnectionIdMap = new ConcurrentHashMap<>();
         this.expirationTimer = new SystemTimerReaper(CLIENT_METRICS_REAPER_THREAD_NAME, new SystemTimer("client-metrics"));
         this.clientTelemetryMaxBytes = clientTelemetryMaxBytes;
         this.time = time;
         this.cacheExpiryMs = cacheExpiryMs;
         this.lastCacheErrorLogMs = new AtomicLong(0);
         this.metrics = metrics;
         this.clientMetricsStats = new ClientMetricsStats();
+        this.connectionDisconnectListener = new ClientConnectionDisconnectListener(clientInstanceCache, clientConnectionIdMap, clientMetricsStats);
```

Comment:
ClientConnectionDisconnectListener is a private class and can use all members in the parent class. Do we need to pass in clientInstanceCache, clientConnectionIdMap, clientMetricsStats?

Reply:
Agree, moved the class to non-static and removed these.